cdap-common/src/main/java/co/cask/cdap/common/conf/Constants.java

```diff
@@ -719,8 +722,37 @@
     public static final String LOGIN_MODULE_CLASS_NAME = "security.authentication.loginmodule.className";
     /** Realm file for Basic Authentication */
     public static final String BASIC_REALM_FILE = "security.authentication.basic.realmfile";
-    /** Enables SSL */
+    /** Enables external SSL */
+    @Deprecated
     public static final String SSL_ENABLED = "ssl.enabled";
+    /** Key to mark a discoverable which supports ssl */
+    public static final String SSL_DISCOVERABLE_KEY = "ssl";
+
+    /**
+     * App Fabric
+     */
+    public static final class Ssl {
+      /** Enables SSL for external services. */
+      public static final String SSL_EXTERNAL_ENABLED = "ssl.external.enabled";
```

Comment:
For the variable name, it shouldn't need `SSL` as prefix as it's already inside the inner class `SSL`.